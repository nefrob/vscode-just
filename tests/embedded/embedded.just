# Embedded languages

python:
  #!/usr/bin/env python3
  print('Hello from python!')

js:
  #!/usr/bin/env node
  console.log('Greetings from JavaScript!')

perl:
  #!/usr/bin/env perl
  print "Larry Wall says Hi!\n";

sh:
  #!/usr/bin/env sh
  hello='Yo'
  echo "$hello from a shell script!"

ruby:
  #!/usr/bin/env ruby
  puts "Hello from ruby!"