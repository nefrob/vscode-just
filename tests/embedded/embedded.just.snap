># Embedded languages
#^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
># TODO: how to test embedded grammars in snapshots?
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>python:
#^^^^^^ source.just entity.name.function.just
#      ^ source.just keyword.operator.recipe.end.just
>  #!/usr/bin/env python3
#^^^^ source.just
#    ^ source.just keyword.operator.path-join.just
#     ^^^ source.just
#        ^ source.just keyword.operator.path-join.just
#         ^^^ source.just
#            ^ source.just keyword.operator.path-join.just
#             ^^^ source.just support.function.builtin.just
#                ^^^^^^^^^ source.just
>  print('Hello from python!')
#^^^^^^^ source.just
#       ^ source.just
#        ^ source.just string.quoted.single.just string.quoted.single.just
#         ^^^^^^^^^^^^^^^^^^ source.just string.quoted.single.just
#                           ^ source.just string.quoted.single.just
#                            ^ source.just
>
>js:
#^^ source.just entity.name.function.just
#  ^ source.just keyword.operator.recipe.end.just
>  #!/usr/bin/env node
#^^^^ source.just
#    ^ source.just keyword.operator.path-join.just
#     ^^^ source.just
#        ^ source.just keyword.operator.path-join.just
#         ^^^ source.just
#            ^ source.just keyword.operator.path-join.just
#             ^^^ source.just support.function.builtin.just
#                ^^^^^^ source.just
>  console.log('Greetings from JavaScript!')
#^^^^^^^^^^^^^ source.just
#             ^ source.just
#              ^ source.just string.quoted.single.just string.quoted.single.just
#               ^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just string.quoted.single.just
#                                         ^ source.just string.quoted.single.just
#                                          ^ source.just
>
>perl:
#^^^^ source.just entity.name.function.just
#    ^ source.just keyword.operator.recipe.end.just
>  #!/usr/bin/env perl
#^^^^ source.just
#    ^ source.just keyword.operator.path-join.just
#     ^^^ source.just
#        ^ source.just keyword.operator.path-join.just
#         ^^^ source.just
#            ^ source.just keyword.operator.path-join.just
#             ^^^ source.just support.function.builtin.just
#                ^^^^^^ source.just
>  print "Larry Wall says Hi!\n";
#^^^^^^^^ source.just
#        ^ source.just string.quoted.double.just string.quoted.double.just
#         ^^^^^^^^^^^^^^^^^^^ source.just string.quoted.double.just
#                            ^^ source.just string.quoted.double.just constant.character.escape.just
#                              ^ source.just string.quoted.double.just
#                               ^^ source.just
>
>sh:
#^^ source.just entity.name.function.just
#  ^ source.just keyword.operator.recipe.end.just
>  #!/usr/bin/env sh
#^^^^ source.just
#    ^ source.just keyword.operator.path-join.just
#     ^^^ source.just
#        ^ source.just keyword.operator.path-join.just
#         ^^^ source.just
#            ^ source.just keyword.operator.path-join.just
#             ^^^ source.just support.function.builtin.just
#                ^^^^ source.just
>  hello='Yo'
#^^^^^^^^ source.just
#        ^ source.just string.quoted.single.just string.quoted.single.just
#         ^^ source.just string.quoted.single.just
#           ^ source.just string.quoted.single.just
>  echo "$hello from a shell script!"
#^^^^^^^ source.just
#       ^ source.just string.quoted.double.just string.quoted.double.just
#        ^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just string.quoted.double.just
#                                   ^ source.just string.quoted.double.just
>
>ruby:
#^^^^ source.just entity.name.function.just
#    ^ source.just keyword.operator.recipe.end.just
>  #!/usr/bin/env ruby
#^^^^ source.just
#    ^ source.just keyword.operator.path-join.just
#     ^^^ source.just
#        ^ source.just keyword.operator.path-join.just
#         ^^^ source.just
#            ^ source.just keyword.operator.path-join.just
#             ^^^ source.just support.function.builtin.just
#                ^^^^^^ source.just
>  puts "Hello from ruby!"
#^^^^^^^ source.just
#       ^ source.just string.quoted.double.just string.quoted.double.just
#        ^^^^^^^^^^^^^^^^ source.just string.quoted.double.just
#                        ^ source.just string.quoted.double.just